src/couch_views/src/couch_views_fdb.erl

```diff
@@ -51,18 +103,31 @@ get_update_seq(TxDb, #mrst{sig = Sig}) ->
 
     case erlfdb:wait(erlfdb:get(Tx, seq_key(DbPrefix, Sig))) of
         not_found -> <<>>;
+        <<51:8, SeqBin:12/binary>> -> fabric2_util:to_hex(SeqBin);
```

Comment:
You can use these helpers to convert from the versionstamp to update_seq for comparisons:

https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/fabric/src/fabric2_fdb.erl#L63-L64

Also as I figured out down below the reason for the strangeness is the overloading of the seq_key field which should be changed.

Reply:
The issue here is that the versionstamp/seq passed to the mango_indexer isn't stored yet and will only be created when the transaction is committed. So we cannot create an update_seq from it. The overload below is because I'm trying to keep the couch_views_fdb api consistent externally.